kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java

```diff
@@ -343,6 +358,14 @@ TopicPartitionWriter getTopicPartitionWriter(TopicPartition tp) {
     return topicPartitionWriters.get(tp);
   }
 
+  private void flushToS3(TopicPartition tp) {
+    try {
+      topicPartitionWriters.get(tp).commitFiles();
+    } catch (Exception e) {
+      log.error("Error flushing {} local data into S3", tp, e);
```

Comment:
just make sure that `TopicPartition` has meaningful `toString()`, so the log message will be valuable.

Reply:
Yes, it has